src/legacy/core_plugins/data/public/search/aggs/param_types/field.test.ts

```diff
@@ -110,10 +108,7 @@ describe('Field', () => {
       const aggParam = new FieldParamType({
         name: 'field',
         type: 'field',
-        filterFieldTypes: (aggConfig: IMetricAggConfig) =>
-          get(aggConfig.schema, 'aggSettings.top_hits.allowStrings', false)
-            ? '*'
-            : KBN_FIELD_TYPES.NUMBER,
+        filterFieldTypes: () => KBN_FIELD_TYPES.NUMBER,
```

Comment:
Make it as function become redundant.
In such a case, `filterFieldTypes` type could get rid of `function` type, 
and the `getAvailableFields` could be simplified:

![image](https://user-images.githubusercontent.com/31325372/75983019-c3e9c800-5ef8-11ea-8757-09285c4453f7.png)